src/planner/rules/agg.rs

```diff
@@ -0,0 +1,157 @@
+use super::*;
+
+/// Returns all rules of aggregation extraction.
+#[rustfmt::skip]
+pub fn rules() -> Vec<Rewrite> { vec![
+    rw!("select-to-plan";
+        "(select ?exprs ?from ?where ?groupby ?having ?orderby ?limit ?offset)" =>
+        "
+        (limit ?limit ?offset
+        (order ?orderby
+        (filter ?having
+        (projagg ?exprs ?groupby
+        (filter ?where
+        ?from
+        )))))"
+    ),
+    rw!("split-projagg";
+        "(projagg ?exprs ?groupby ?child)" =>
+        { ExtractAgg {
+            has_agg: pattern("(proj ?exprs (agg ?aggs ?groupby ?child))"),
```

Comment:
There could be agg in having clause. Maybe we will need more test cases to cover this.

Reply:
You are right. There is a bug here. 🥵

Reply:
Fixed in b8edc65. PTAL!

Reply:
Still not correct. The HAVING expression should be extracted as well.
Thanks for your great article. 📖

Reply:
fixed in #706